### PR TITLE
Feature/cli

### PR DESCRIPTION
**Pull request by HomelessSkittle**
CLI should be updated to now work with fields and methods. Output also slightly changed to reflect this.

Contains changes made to Classes.java to prevent multiple relationships between classes
Also contains all tests merged to develop as of 9/17/20

### CODE REVIEW

**Comment by HomelessSkittle** [comment]
As a heads-up/warning/PSA/to whom it may concern:

There were a lot of conflicts that couldn't be resolved (The 'there's a one line difference but github can't figure out what", problem) and I fixed it up to remove those issues. 

This unfortunately makes it look like a lot of things were changed in files that don't actually get touched at all like Classes.java or Relationships.java.

File: src/datastructures/Classes.java [hunk 1]
```diff
@@ -288,3 +288,5 @@ public Relationship getRelationship(String name, String class1, String class2)
     }
 
 }
+
+}
```
Comment (SerbianSensation) [inline src/datastructures/Classes.java:292]: Added an extra curly brace at the end.

File: src/testcases/RelationshipTest.java [hunk 2]
```diff
@@ -44,4 +44,3 @@ public void testEquals()
     	assertFalse("Differing first classes should result in false", r1.equals(r4));
     	assertFalse("Differing second classes should result in false", r1.equals(r5));
     }
-}
```
Comment (SerbianSensation) [inline src/testcases/RelationshipTest.java:47]: Missing a curly brace to close the Class.

File: src/datastructures/Relationship.java [hunk 3]
```diff
@@ -68,4 +68,3 @@ public boolean equals(Object obj) {
 		}
 		return false;
 	}
-}
```
Comment (SerbianSensation) [inline src/datastructures/Relationship.java:71]: Missing a curly brace to close the Class.

File: src/testcases/EntityTest.java [hunk 4]
```diff
@@ -151,4 +151,4 @@ public void testDeleteMethod()
         assertEquals("List size is 1", 1, e.getMethods().size());
         assertTrue("a2 method still exists", e.getMethods().contains("a2"));
     }
-}
\ No newline at end of file
+
```
Comment (SerbianSensation) [inline src/testcases/EntityTest.java]: Missing a curly brace to close the Class.

**Review by SerbianSensation — CHANGES_REQUESTED** [review]
Command Line part looks good, but there are some missing/extra curly braces in the datastructures/testcases files. Don't want the program to crash because of those annoying little things! ;)

File: src/testcases/ClassesTest.java [hunk 5]
```diff
@@ -1,454 +1,454 @@
...
-     */
-    @Test
-    public void testClear()
-    {
-        Classes classes = new Classes();
-        classes.createClass("Entity1");
-        classes.createClass("Entity2");
-        classes.createRelationship("Linkage", "Entity1", "Entity2");
-
-        classes.clear()
```
Comment (JohnHynes) [inline src/testcases/ClassesTest.java:49]: Missing ;

File: src/testcases/ClassesTest.java [hunk 6]
```diff
@@ -1,454 +1,454 @@
...
-        classes.createField("E1", "att1");
-        classes.createField("E1", "att2");
-        classes.createMethod("E1", "m1");
-        classes.createMethod("E1", "m2");
-
-        Entity new_e1copy = classes.copyEntity("E1");
-        assertTrue("Entity with fields and Methods was copied correctly", classes.getEntities().get(0).equals(new_e1copy));
-        assertTrue("Fields are the same", classes.getEntities().get(0).getFields().equals(new_e1copy.getFields()));
-        assertTrue("Methods are the same", classes.getEntities().get(0).getMethods().equals(new_e1copy.getMethods()));
-        assertFalse("Old e1_copy no longer equal to E1", classes.getEntities().get(0).equals(e1copy))
```
Comment (JohnHynes) [inline src/testcases/ClassesTest.java:115]: Missing ;

File: src/cli/CommandLine.java [hunk 7]
```diff
@@ -1,166 +1,248 @@
...
+						}
+					} else {
+						System.out.println(errorMessage + commandUsage[10] + commandUsage[11] + commandUsage[12] + "\n");
+					}
+					break;
+				// Call list class or relationship based on length and user input
+				case "list":				
+					if (commands[1].equals("classes")) {
+						if (commands.length != 2) {
+							System.out.println(errorMessage + commandUsage[113] + "\n");
```
Comment (JohnHynes) [inline src/cli/CommandLine.java:159]: 13 instead of 113?

**Review by JohnHynes — CHANGES_REQUESTED** [review]
Looks good, a couple of small things I found but otherwise, it's looking pretty finished.

File: src/testcases/ClassesTest.java (hunk 5 above)
Comment (HomelessSkittle) [reply src/testcases/ClassesTest.java:49]: Fixed

File: src/testcases/ClassesTest.java (hunk 6 above)
Comment (HomelessSkittle) [reply src/testcases/ClassesTest.java:115]: Fixed

File: src/cli/CommandLine.java (hunk 7 above)
Comment (HomelessSkittle) [reply src/cli/CommandLine.java]: Fixed

**Review by JohnHynes — APPROVED** [review]
I haven't found anything else, looks good!

**Review by SerbianSensation — APPROVED** [review]
Looks good to me. Good job fixing the curly braces, and the hard to spot missing semicolon. I like the way you set up the command usage messages, the indexes are easy to follow since they are in order down the CLI and can be re-used in the help messages. Very nice.

**Pull request merged**